lib/ttt.ex

```diff
@@ -5,55 +5,133 @@ defmodule TTT do
   end
 
   def new do
-    IO.puts "Let's play Tic Tac Toe!"
+    IO.puts "Let's play Tic Tac Toe!\n"
+    input = IO.gets "Please select a game mode: \n\n1. Player vs. Player\n2. Player vs. Computer\n3. Spectate a game\n\nYour choice is: "
+    String.trim(input)
+    |> game_mode()
+    |> start()
+  end
+
+  def game_mode(choice) do
+    cond do
+      choice == "1" -> player_player()
+      choice == "2" -> player_comp()
+      choice == "3" -> comp_comp()
+    end
+  end
+
+  def start({player1, player2} = _players) do
     board = Board.new
-    game = Game.new
+    game = Game.new(player1, player2)
```

Comment:
Having code like `Game.new` or `Board.new` is reminiscent of object-oriented style with classes and constructors. Although there is nothing wrong with this, per say, it does seem out of place in a functional paradigm. Instead of "new", why not try `Board.empty_board` or something similar? Remember, these are functions, not methods on a class.